app/assets/js/app/modules/totalliser.js

```diff
@@ -0,0 +1,75 @@
+import domready from './domready'
+import {forEach} from 'lodash'
+import getTransitionEndEvent from '../helpers/transitionend'
+
+const transitionendEvent = getTransitionEndEvent()
+
+domready(() => {
+  const breakdownFields = document.querySelectorAll('input:not(.js-input--calculated)')
+  const totalField = document.querySelector('.js-input--calculated')
+
+  const onAnswerChanged = () => {
+    let newTotal = 0
+
+    for (let i = 0; i < breakdownFields.length; i++) {
+      let percentage = breakdownFields[i].value
+
+      let value = parseInt(percentage)
+      if (isNaN(value)) {
+        continue
+      } else if (value > 0) {
+        newTotal += value
+      }
+    }
+
+    updateTotal(newTotal)
+  }
+
+  const highlight = () => {
+    if (transitionendEvent) {
+      totalField.addEventListener(transitionendEvent, onTransitionHighlightEnd, true)
+    } else {
+      window.setTimeout(onTransitionHighlightEnd, 500)
+    }
+    totalField.classList.add('input--has-error')
+  }
+
+  const removeHighlight = () => {
+    if (transitionendEvent) {
+      totalField.addEventListener(transitionendEvent, onTransitionHighlightEnd, true)
+    } else {
+      window.setTimeout(onTransitionHighlightEnd, 500)
+    }
+    totalField.classList.remove('input--has-error')
+  }
+
```

Comment:
both of these functions can be combined into a `toggleHighlight` function as they are largely the same apart from the add/remove aspect.